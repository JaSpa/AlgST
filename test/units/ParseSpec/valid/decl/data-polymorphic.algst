data BTree a
  = Branch (BTree a) (BTree a)
  | Leaf a

data BTree2 branch leaf
  = Branch2 branch (BTree2 branch leaf) (BTree2 branch leaf)
  | Leaf2 leaf
