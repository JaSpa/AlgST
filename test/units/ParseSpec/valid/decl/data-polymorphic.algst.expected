imports
|

declarations
|
+- BTree
|  |
|  `- DataDecl
|     |
|     +- kind
|     |  |
|     |  `- TU
|     |
|     +- parameters
|     |  |
|     |  `- (a:TU)
|     |
|     `- constructors
|        |
|        +- Branch
|        |  |
|        |  +- Type.App
|        |  |  |
|        |  |  +- Type.Con BTree
|        |  |  |
|        |  |  `- Type.Var a
|        |  |
|        |  `- Type.App
|        |     |
|        |     +- Type.Con BTree
|        |     |
|        |     `- Type.Var a
|        |
|        `- Leaf
|           |
|           `- Type.Var a
|
+- BTree2
|  |
|  `- DataDecl
|     |
|     +- kind
|     |  |
|     |  `- TU
|     |
|     +- parameters
|     |  |
|     |  +- (branch:TU)
|     |  |
|     |  `- (leaf:TU)
|     |
|     `- constructors
|        |
|        +- Branch2
|        |  |
|        |  +- Type.Var branch
|        |  |
|        |  +- Type.App
|        |  |  |
|        |  |  +- Type.App
|        |  |  |  |
|        |  |  |  +- Type.Con BTree2
|        |  |  |  |
|        |  |  |  `- Type.Var branch
|        |  |  |
|        |  |  `- Type.Var leaf
|        |  |
|        |  `- Type.App
|        |     |
|        |     +- Type.App
|        |     |  |
|        |     |  +- Type.Con BTree2
|        |     |  |
|        |     |  `- Type.Var branch
|        |     |
|        |     `- Type.Var leaf
|        |
|        `- Leaf2
|           |
|           `- Type.Var leaf
|
+- Branch
|  |
|  `- Decl.DataCon (BTree)
|     |
|     +- parameters
|     |  |
|     |  `- (a:TU)
|     |
|     +- multiplicity
|     |  |
|     |  `- U
|     |
|     `- items
|        |
|        +- Type.App
|        |  |
|        |  +- Type.Con BTree
|        |  |
|        |  `- Type.Var a
|        |
|        `- Type.App
|           |
|           +- Type.Con BTree
|           |
|           `- Type.Var a
|
+- Branch2
|  |
|  `- Decl.DataCon (BTree2)
|     |
|     +- parameters
|     |  |
|     |  +- (branch:TU)
|     |  |
|     |  `- (leaf:TU)
|     |
|     +- multiplicity
|     |  |
|     |  `- U
|     |
|     `- items
|        |
|        +- Type.Var branch
|        |
|        +- Type.App
|        |  |
|        |  +- Type.App
|        |  |  |
|        |  |  +- Type.Con BTree2
|        |  |  |
|        |  |  `- Type.Var branch
|        |  |
|        |  `- Type.Var leaf
|        |
|        `- Type.App
|           |
|           +- Type.App
|           |  |
|           |  +- Type.Con BTree2
|           |  |
|           |  `- Type.Var branch
|           |
|           `- Type.Var leaf
|
+- Leaf
|  |
|  `- Decl.DataCon (BTree)
|     |
|     +- parameters
|     |  |
|     |  `- (a:TU)
|     |
|     +- multiplicity
|     |  |
|     |  `- U
|     |
|     `- items
|        |
|        `- Type.Var a
|
`- Leaf2
   |
   `- Decl.DataCon (BTree2)
      |
      +- parameters
      |  |
      |  +- (branch:TU)
      |  |
      |  `- (leaf:TU)
      |
      +- multiplicity
      |  |
      |  `- U
      |
      `- items
         |
         `- Type.Var leaf

